### DISCUSSION TITLE
i want to deploy privategpt on my macos but i meet trouble with 'ggml_metal_init'

### DISCUSSION THREAD

**Discussion by xuhongxin** [imartinez/privateGPT#1654] [General]
1. the mannel i reference is : https://docs.privategpt.dev/installation and https://gist.github.com/mberman84/9b3c281ae5e3e92b7e946f6a09787cde
2. when i run ‘PGPT_PROFILES=local make run’ in the vscode, there is a long error report: 
![image](https://github.com/imartinez/privateGPT/assets/1044109/9af5c088-960a-49bd-8269-5f09349ade7c)
`
ggml_metal_init: allocating
ggml_metal_init: found device: AMD Radeon Pro 5300M
ggml_metal_init: found device: Intel(R) UHD Graphics 630
ggml_metal_init: picking default device: AMD Radeon Pro 5300M
ggml_metal_init: default.metallib not found, loading from source
ggml_metal_init: GGML_METAL_PATH_RESOURCES = nil
ggml_metal_init: loading '/Users/xuhongxin/Library/Caches/pypoetry/virtualenvs/private-gpt-NdiI2ph1-py3.11/lib/python3.11/site-packages/llama_cpp/ggml-metal.metal'
ggml_metal_init: error: Error Domain=MTLLibraryErrorDomain Code=3 "program_source:58:9: error: invalid type 'const constant int64_t &' (aka 'const constant long &') for buffer declaration
        constant  int64_t & ne00,
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:58:19: note: type 'int64_t' (aka 'long') cannot be used in buffer pointee type
        constant  int64_t & ne00,
                  ^
program_source:59:9: error: invalid type 'const constant int64_t &' (aka 'const constant long &') for buffer declaration
        constant  int64_t & ne01,
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:59:19: note: type 'int64_t' (aka 'long') cannot be used in buffer pointee type
        constant  int64_t & ne01,
`
3. my poetry enviroment info is
![image](https://github.com/imartinez/privateGPT/assets/1044109/7002b2a8-5ef5-47b7-b7a4-770b06446903)

**Comment by jp555soul** [comment]
@xuhongxin Did you find a solution?
